MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/BuyClientDefault.java

```diff
@@ -496,32 +497,32 @@ public CancellableTask getProducts(List<Long> productIds, Callback<List<Product>
     }
 
     @Override
-    public CancellableTask getProducts(int page, Long collectionId, Callback<List<Product>> callback) {
-        return productService.getProducts(page, collectionId, callback);
+    public CancellableTask getCollections(int page, Callback<List<Collection>> callback) {
+        return productService.getCollections(page, callback);
     }
 
     @Override
-    public Observable<List<Product>> getProducts(int page, Long collectionId) {
-        return productService.getProducts(page, collectionId);
+    public Observable<List<Collection>> getCollections(int page) {
+        return productService.getCollections(page);
     }
 
     @Override
-    public CancellableTask getProducts(int page, Long collectionId, SortOrder sortOrder, Callback<List<Product>> callback) {
-        return productService.getProducts(page, collectionId, sortOrder, callback);
+    public CancellableTask getProductTags(int page, Callback<List<String>> callback) {
+        return productService.getProductTags(page, callback);
     }
 
     @Override
-    public Observable<List<Product>> getProducts(int page, Long collectionId, SortOrder sortOrder) {
-        return productService.getProducts(page, collectionId, sortOrder);
+    public Observable<List<String>> getProductTags(int page) {
+        return productService.getProductTags(page);
     }
 
     @Override
-    public CancellableTask getCollections(int page, Callback<List<Collection>> callback) {
-        return productService.getCollections(page, callback);
+    public CancellableTask getProducts(int page, Long collectionId, Set<String> tags, SortOrder sortOrder, Callback<List<Product>> callback) {
+        return productService.getProducts(page, collectionId, tags, sortOrder, callback);
     }
 
     @Override
-    public Observable<List<Collection>> getCollections(int page) {
-        return productService.getCollections(page);
+    public Observable<List<Product>> getProducts(int page, Long collectionId, Set<String> tags, SortOrder sortOrder) {
```

Comment:
I find this overloading of the api a bit confusing.  I can see that tags can be used to filter if have/don't have a collection Id, but sortOrder is only used if collectionId is not null.
For clarity of the API I would prefer to see a bit of code duplication with two APIs, one with (page, collectionId, tags, sortOrder) where collectionId is mandatory and one with (page, tags) that gets all products filtered by  tags.
@bgulanowski @davidmuzi ?